lib/bundle/cask_dumper.rb

```diff
@@ -13,15 +13,33 @@ def casks
 
       require "cask/caskroom"
 
-      @casks ||= Cask::Caskroom.casks.map(&:to_s)
-      @casks.map { |cask| cask.chomp " (!)" }
-            .uniq
+      @casks ||= Cask::Caskroom.casks.map(&:to_s).uniq
+    end
+
+    def casks_config
+      return {} unless Bundle.cask_installed?
+
+      @casks_config ||= Hash[Cask::Caskroom.casks.map(&:to_s).zip(
+        Cask::Caskroom.casks.map do |cask|
+          if cask.config.nil? || cask.config.explicit.blank?
+            {}
+          else
+            cask.config.explicit.map do |k, v|
+              "#{k}: \"#{v.sub(/^#{ENV['HOME']}(#{File::SEPARATOR}|$)/, "~/")}\""
+            end.join(",")
+          end
+        end,
+      )]
     end
 
     def dump(casks_required_by_formulae)
       [
-        (casks & casks_required_by_formulae).map { |cask| "cask \"#{cask}\"" }.join("\n"),
-        (casks - casks_required_by_formulae).map { |cask| "cask \"#{cask}\"" }.join("\n"),
+        (casks & casks_required_by_formulae).map do |cask|
+          "cask \"#{cask}\"" + (casks_config[cask].blank? ? "" : ", args: { #{casks_config[cask]} }")
```

Comment:
`casks_config[cask].blank? ? "" : ", args: { #{casks_config[cask]} }"` is a bit of a mouthful and is repeated; would be nice to move it into a separate method if possible.

Reply:
Simplified it a bit.